packages/monosize-storage-azure/src/uploadReportToRemote.test.mts

```diff
@@ -82,6 +103,41 @@ describe('uploadReportToRemote', () => {
     const localReport = sampleReport.slice(0, 1);
 
     getRemoteReport.mockReturnValueOnce(remoteReport);
+    const uploadReportToRemote = createUploadReportToRemote(testConfig);
+    await uploadReportToRemote(branchName, commitSHA, localReport);
+
+    expect(submitTransaction).toHaveBeenCalledTimes(1);
+    expect(submitTransaction).toHaveBeenCalledWith([
+      ['delete', { partitionKey: 'main', rowKey: createRowKey(sampleReport[1]) }],
+      [
+        'upsert',
+        { ...sampleReport[0], commitSHA, partitionKey: 'main', rowKey: createRowKey(sampleReport[0]) },
+        'Replace',
+      ],
+    ]);
+  });
+
+  it('uploads a report to the remote using AzurePipelinesCredential authType', async () => {
+    // Remote report contains 2 entries, local report contains 1 entry
+    // => 1 entry should be deleted, 1 entry should be upserted
+
+    Object.assign(process.env, {
+      AZURE_TENANT_ID: 'azure-tenant-id',
+      AZURE_CLIENT_ID: 'azure-client-id',
+      AZURE_SERVICE_CONNECTION_ID: 'azure-service-connection-id',
+      SYSTEM_ACCESSTOKEN: 'system-access-token',
+    });
```

Comment:
I am not a big fan of this 🐱  To clarify: any _next executed_ test will have these env variables that could make tests fragile and order dependent.

To avoid it, I suggest:
- move out `getTableClient()` (and probably rename to `createTableClient()`, too) to a separate file
- add tests there to assert `AzurePipelinesCredential` creation
- in this file just mock `getTableClient()` and ensure that it's called with proper config

Reply:
good thinking 👆🏻, 

not the biggest fan of exposing private functionality for sake of a test, so here are couple of other options:

- 1. extend storage config 

```
export type AzureStorageConfig = {
  endpoint: string;
  /**
   * @default 'AzureNamedKeyCredential'
   */
  authType?: AzureAuthenticationType;

👉  environment?: Record<Keys_We_Use, string>;
};
```

with this approach user could set those values directly within config ( in a template style of format  to not leak 

- 2. extend arity of the function for internal purposes

```
export function createUploadReportToRemote(
  config: AzureStorageConfig, 
👉  __env: Record<string, string>) {
```

Reply:
I refactored to rename `getTableClient` to be `createTableClient` and moved `createTableClient` to its own file with its own corresponding test file. Let me know if I'm missing any test cases that I should cover :) [470bf7c](https://github.com/microsoft/monosize/pull/79/commits/470bf7cda5eea5dd7556415bc3bdc48bd59a82a8)